### DISCUSSION TITLE
Thoughts, Metadata and Templates

### DISCUSSION THREAD

**Discussion by nianiam** [notaapp/talk#102] [Feature Requests] [answered]
For quite a while I was using Obsidian. It does a lot of things right, but had a few things wrong that were dealbreakers for me. What Nota does, it does _right_, which is a breath of fresh air and I am enjoying the experience so far.

That being said, Nota is somewhat 'behind' on features many other markdown apps offer - especially when some are free (Obsidian, Notable etc). There are already issues discussing many of my smaller gripes (piped links for example), but for me the 'critical' missing features are:

- YAML metadata, or just tag support. I actually prefer folder organisation (for which I love how Nota shows all sub-folder notes when you select a folder), but I do miss the ability to create tags to cross-link specific types of notes. This [issue](https://github.com/notaapp/nota/issues/86) has no responses 😢.
- Templates. Obsidian's vanilla implementation should be the standard, but the community plugin 'Templater' allowing a variety of variables is amazing.

Neither are in the roadmap, but in my opinion would improve the app more significantly.

**Comment by astoilkov** [comment, marked as answer]
HI @nianiam,

Thanks for the kind words. Let me get straight to answering your questions:
- **Tags.** Tags are something we are currently exploring. Nota tries to do things *right* as you said it. This requires more time as we iterate and experiment a lot. In general, when we know the correct UX the implementation can happen in a few days. Do you have a favorite implementation of tags somewhere?
- **YAML metadata.** What do you mean by that exactly? We support YAML at the top of the file. Do you imagine something extra on top of that?
- **Regarding the [issue](https://github.com/notaapp/nota/issues/86) you pointed out.** We have discussed it internally several times and will respond to it. We have talked to the user several times in other discussions and this is why we don't respond immediately (but maybe we should).
- **Templates.** This is something we haven't researched enough. We will research it more and get back to you. Do you have some features that you consider a must?

**Reply by nianiam** [reply]
Hi, thanks for the response!

- I wasn't very clear on my first point, sorry. I was only really saying that Nota should aim to use both YAML and inline tags, as suggested by the issue I linked. However, a response as to planned features around metadata (or just tags!) and how they might be able to leveraged might be good.
- As for a UX implementation of tags, most apps are the same: a pane with a tree of nested tags. As long as they're nested and filterable in some way I would be happy with the same.
- Regarding templates, I liked Obsidian's implementation. You make a folder of template files, then use a shortcut to insert the contents of that template file to the cursor in the current file or open the template as a new file. The ability to load variables (time, date, folder name etc) is a must though.

Tags would probably align Nota with most of the other apps out there. Templates, push it above! Keep up the good work.

**Reply by astoilkov** [reply]
> I wasn't very clear on my first point, sorry. I was only really saying that Nota should aim to use both YAML and inline tags, as suggested by the issue I linked.

By YAML tags you mean something like this?:
```
---
tags: metadata, tags, yaml
---
```

By inline tags do you mean `#metadata #inline-tags`?

> However, a response as to planned features around metadata (or just tags!) and how they might be able to leverage might be good.

Currently, I can't give an estimation around tags. We are exploring and spending time every week experimenting and exploring it.

> Regarding templates, I liked Obsidian's implementation. You make a folder of template files, then use a shortcut to insert the contents of that template file to the cursor in the current file or open the template as a new file. The ability to load variables (time, date, folder name etc) is a must though.

Thanks for the additional information. Have you ever considered using a snippets app to insert templates? This way you will have a solution that works in every app not just the app that supports it. Here are few possible apps you can use:
- [TextExpander – Productivity App | Text Expander](https://textexpander.com/) is probably the most popular way to solve this problem. Writing a specific phrase will insert the associated snippet for this phrase. You can have a snippet like `;today` that will insert the today's date.
- [Alfred](https://www.alfredapp.com/). If you are already a paying user of Alfred you can use their [Snippets](https://www.alfredapp.com/help/features/snippets/) feature to quickly add snippets in any app.
- [Raycast](https://raycast.com/). Use the Clipboard History feature of Raycast. Assign a shortcut to the Clipboard History to access it with a single shortcut and then pin items(using ⌘K) that will act as snippets. The bonus is that you will have the power of clipboard history.

**Reply by nianiam** [reply]
Yes, those are the tag syntaxes I would use.

Regarding snippets, I use Raycast and it's great. I use their snippets for certain things, but the advantage of Obsidian's template system was that you could add variables in those snippets. These included more general things like date and time, but also the path the current file or folder location. Templater (the community plugin) also allowed you to write simple JS scripts to do more with Node's path class (and anything else for that matter). As you can imagine this synergises very well with folder based organisation.

Thanks for the transparency around these topics (and advice on the other apps). I hope these points are useful. In my opinion metadata _should_ be something Nota offers given the competition, however, I appreciate there are many ways to do it and I'm sure they not a simple addition.

**Reply by astoilkov** [reply]
> Regarding snippets, I use Raycast and it's great. I use their snippets for certain things, but the advantage of Obsidian's template system was that you could add variables in those snippets. These included more general things like date and time, but also the path the current file or folder location. Templater (the community plugin) also allowed you to write simple JS scripts to do more with Node's path class (and anything else for that matter). As you can imagine this synergises very well with folder based organisation.

Great! That makes sense. Do you have a template that you are comfortable sharing? I am trying to understand how do you use the current file, folder location, or the type of JS scripts you have. If you agree, you can send it to contact@nota.md.

> Thanks for the transparency around these topics (and advice on the other apps). I hope these points are useful. In my opinion metadata should be something Nota offers given the competition, however, I appreciate there are many ways to do it and I'm sure they not a simple addition.

Thank you for the understanding.
